src/cbc_sdk/endpoint_standard/base.py

```diff
@@ -616,3 +697,297 @@ def _search(self, start=0, rows=0):
 
             log.debug("current: {}, total_results: {}".format(current, self._total_results))
 
+
+class EnrichedEventFacetQuery(Query, AsyncQueryMixin):
+    """Represents the query logic for an Enriched Event Facet query.
+    This class specializes `Query` to handle the particulars of
+    enriched events faced querying.
+    """
+    def __init__(self, doc_class, cb):
+        super(EnrichedEventFacetQuery, self).__init__(doc_class, cb)
+        self._ranges = []
+        self._default_args = {}
+        self._query_token = None
+        self._timeout = 0
+        self._timed_out = False
+        self._time_range = {}
+        self._limit = 100
+        self._facet_fields = []
+        self._facet_rows = 100
+        self._query_builder = QueryBuilder()
+
+    def or_(self, **kwargs):
+        """ or_ criteria are explicitly provided to EnrichedEvent queries although they are endpoint_standard.
+            This method overrides the base class in order to provide or_() functionality rather than
+            raising an exception
+        """ 
+        self._query_builder.or_(None, **kwargs)
+        return self
+        
+
+    def _get_query_parameters(self):
+        """Need to override base class implementation as it sets custom (invalid) fields"""
+        args = self._default_args.copy()
+        if not self._facet_fields:
```

Comment:
I just found that it's OK to have `_facet_fields` undefined as long as `_ranges` is defined, so I would just switch this check to `if not (self._facet_fields or self._ranges):`

Reply:
Indeed. Changed.